src/test/java/seedu/address/logic/LogicManagerTest.java

```diff
@@ -147,16 +141,38 @@ private void assertCommandFailure(String inputCommand, Class<? extends Throwable
     }
 
     /**
-     * A stub class to throw an {@code IOException} when the save method is called.
+     * Execute a dummy command in a {@code LogicManager} containg a {@code Storage} Stub that will
+     * throw an {@code IOException}, this will confirm that <br>
+     * - the {@code expectedException} is thrown <br>
+     * - the resulting error message is equal to {@code expectedMessage} <br>
+     *
+     * @param e exception to be thrown by the AddressBookStorage
+     * @param expectedMessage expected message to be thrown
      */
-    private static class JsonAddressBookIoExceptionThrowingStub extends JsonAddressBookStorage {
-        private JsonAddressBookIoExceptionThrowingStub(Path filePath) {
-            super(filePath);
-        }
-
-        @Override
-        public void saveAddressBook(ReadOnlyAddressBook addressBook, Path filePath) throws IOException {
-            throw DUMMY_IO_EXCEPTION;
-        }
+    private void assertCommandFailureForException(IOException e, String expectedMessage) {
```

Comment:
Make the name a bit more specific?
```suggestion
    private void assertCommandFailureForExceptionFromStorage(IOException e, String expectedMessage) {
```